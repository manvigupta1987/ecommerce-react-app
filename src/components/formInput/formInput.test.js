
import React from 'react';
import { shallow } from 'enzyme';
import {FormInput} from './formInput.component'

describe('FormInput component testing', () => {
  let mockHandleChange;
  let wrapper;

  beforeEach(()=>{
    mockHandleChange = jest.fn();

    const mockProps = {
      label: 'email',
      value: '123@gmail.com',
      handleChange: mockHandleChange
    }

    wrapper = shallow(<FormInput {...mockProps} />)
  });

  it('FormInput Snapshot testing', ()=> {
    expect(wrapper).toMatchSnapshot();
  })

  it('expect to render Label when label is present', ()=> {
    expect(wrapper.exists('FormInputLabel')).toBe(true);
  })

  it('expect to render Nothing when label is not present', ()=> {
    const mockProps = {
      label: '',
      value: '123@gmail.com',
      handleChange: mockHandleChange
    }

    const wrapper = shallow(<FormInput {...mockProps} />)
    expect(wrapper.exists('FormInputLabel')).toBe(false);
  })

  it('expect to get called handleChange function when click happens', ()=> {
    wrapper.find('FormInputContainer').simulate('change')
    expect(mockHandleChange).toHaveBeenCalled();
  })
})