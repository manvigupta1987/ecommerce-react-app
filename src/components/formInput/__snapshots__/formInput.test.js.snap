// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component testing FormInput Snapshot testing 1`] = `
<styled.div>
  <FormInputContainer
    onChange={[MockFunction]}
    value="123@gmail.com"
  />
  <FormInputLabel
    className="shrink"
  >
    email
  </FormInputLabel>
</styled.div>
`;
